Project1/memory_manager.cpp

```diff
@@ -0,0 +1,76 @@
+#include <stdlib.h>
+#include <iostream>
+
+#include "memory_manager.h"
+
+MemoryBlock* first_block = nullptr;
+
+MemoryBlock* new_mem_block(size_t size)
+{
+	MemoryBlock* mem = static_cast<MemoryBlock*>(malloc(sizeof(MemoryBlock) + size));
+	if (mem == nullptr) {
+		throw std::bad_alloc();
+	}
+	mem[0].is_allocated = true;
```

Comment:
You can (and should) use mem->is_allocated, which is equivalent to accessing the first element in an array (like you're doing) except logically mem isn't a pointer to an array, it's a pointer to a single object residing on the heap, and therefore the use of '->' is more appropriate.